src/jazkarta/zoterolib/content/zotero_library.py

```diff
@@ -42,6 +42,26 @@ def index_element(self, element):
         catalog = api.portal.get_tool("portal_catalog")
         catalog.catalog_object(obj, uid=obj.path)
 
+    def fetch_items(self, start=0, limit=100):
+        """Iterates over ALL remote items, starting at the given offset.
+        The limit will be used to determine how many items to retrieve at once.
+        """
+        zotero_api = zotero.Zotero(self.zotero_id, self.zotero_library_type)
+
+        current_batch = zotero_api.top(start=start, limit=limit)
```

Comment:
Do we want `top(...)` or `items(...)` here? I think `top(...)` only makes sense if we want to also fetch `collections_top(...)` and then make potentially nested calls to `collections_sub(collection_id)`. That might be the decent way to get the full structure if/when we care about the structure metadata and organization, but it's probably less efficient than calling `items(...)` and `all_collections(...)` and generating the structure from that data.

Reply:
I don't really yet understand the nuances of these methods. I focused on making it work to then refine it later. Also the generator based approach is meant for the synchronous use case. My plan was to possibly write another version of the fetch method that accepts start and limit, to be used in the asynchronous case.

Reply:
Sounds good. For now, I'd recommend using `items()` to get everything in the library regardless of folder level and we can worry about structure later. I think with `top()` a library like the ISAW one will return no results since everything is inside of a collection. In cases where we want to replicate the collection structure we can also fetch `all_collections` which will probably need a slightly different data proxy object for indexing. That's something for another story though.

Reply:
I switched to `items()` on my branch(es) approving this for merge now.